src/tests/test_parser.rs

```diff
@@ -523,12 +545,33 @@ fn list_type_annotations_inner_not_type() -> TestResult {
 }
 
 #[test]
-fn list_type_annotations_with_spaces() -> TestResult {
+fn list_type_annotations_inner_original_list() -> TestResult {
+    let input = "def test [list: list<list>] { $list | length }; test [[2 5 4] [a b c]]";
+    let expected = "2";
+    run_test(input, expected)
+}
+
+#[test]
+fn list_type_annotations_unknown_separators() -> TestResult {
+    let input = "def test [list: list<int, string>] { $list | length }; test [2 5 4]";
+    let expected = "use `>` to terminate it";
```

Comment:
Would be great if we could use something like `ParseError::ExtraPositional` or `ExtraToken` here to indicate that it is too much. Don't know if we get to see that from the lexer?

Don't know if the whitespace variants below have to be errors or if we could be tolerant to them?

Reply:
Ah that sucks a bit. I think, if we want to repeat your work for records, we would need to be able to support that, right?